rero_ils/modules/loans/logs/statistics.py

```diff
@@ -0,0 +1,156 @@
+    query = RecordsSearch(index=LoanOperationLog.index_name)[:0] \
+        .query('bool', filter=filters)
+    query.aggs.bucket('operations', agg)
+    results = query.execute()
+
+    return {
+        time_interval.key_as_string: {
+            'total': time_interval.doc_count,
+            'operations': {
+                operation.key: operation.doc_count
+                for operation in time_interval.operation.buckets
+            }
+        }
+        for time_interval in results.aggregations.operations.buckets
+    }
+
+
+def get_rate_circulation_statistics(
+    filters=None, interval=1, operations=None, **kwargs
+):
+    """Get rate statistics (HH:mm) about circulation operations.
+
+    Get circulation statistics filtered with some filters and get the
+    aggregates them by minutes of the day.
+
+    :param filters: a list of criteria to apply to filter the operation
+        logs index.
+    :param interval: the number of minutes between each result step.
+    :param operations: the list of operation name to analyze.
+    :param kwargs: any additional named parameters. Useful key are:
+        * `from` : the transaction lower boundary interval limit (included)
+        * `to`: the transaction upper boundary interval limit (include)
+    :return: a dictionary representing circulation operation by minutes of
+        the day. Each entry key representing the minutes of the day using
+        format `HH:mm`, each entry values is a dictionary where circulation
+        operation/counter are describe.
+
+    >> {
+    >>    '00:01': {'checkin': 0, 'checkout':3, 'renewal':0 },
+    >>    '00:02': {'checkin': 2, 'checkout':5, 'renewal':3 },
+    >>    ...
+    >> }
+    """
+    # 1) Build filters to use to filter the search result
+    # 2) Add histogram aggregation to the query
+    # 3) Execute the query
+    filters = filters or []
+    filters.append(Q('term', record__type='loan'))
+    if date_range := _build_date_filters(kwargs.get('from'), kwargs.get('to')):
+        filters.append(Q('range', date=date_range))
+    agg = A(
+        'histogram',
+        script='doc["date"].value.get(ChronoField.MINUTE_OF_DAY)',
+        min_doc_count=1,
+        interval=interval
+    )
+    agg.bucket('operation', A('terms', field='loan.trigger'))
+    search = RecordsSearch(index=LoanOperationLog.index_name) \
+        .query('bool', filter=filters)
+    search.aggs.bucket('operations', agg)
+    search = search[:0]
+    results = search.execute()
+
+    # Build the default statistic results
+    operations = operations or app.config.get('CIRCULATION_BASIC_OPERATIONS')
+    operations_data = {operation: 0 for operation in operations}
+    stats = {
+        f'{idx // 60:02d}:{idx % 60:02d}': deepcopy(operations_data)
+        for idx in range(0, 1440, interval)
+    }
+
+    # Analyze the ES query response to populate the stats dictionary
+    for aggr in results.aggregations.operations.buckets:
```

Comment:
the aggregations object could has no attribute 'operations'.
you can use something like `results.aggregations.get('operations', {}).get('buckets', [])`